rotkehlchen/globaldb/updates.py

```diff
@@ -300,74 +305,219 @@ def _parse_full_insert(self, insert_text: str) -> AssetData:
+    ) -> None:
+        """
+        Given the already processed information for an asset try to store it in the globaldb
+        and if it is not possible due to conflicts mark it to resolve later.
+        """
+        local_asset: Optional[Asset] = None
+        with suppress(UnknownAsset):
+            local_asset = Asset(remote_asset_data.identifier).check_existence()
+
+        try:
+            with connection.savepoint_ctx() as cursor:
+                executeall(cursor, action)
+        except sqlite3.Error:  # https://docs.python.org/3/library/sqlite3.html#exceptions
+            if local_asset is None:
+                try:  # if asset is not known then simply do an insertion
+                    with connection.savepoint_ctx() as cursor:
+                        executeall(cursor, full_insert)
+                except sqlite3.Error as e:
+                    self.msg_aggregator.add_warning(
+                        f'Failed to add asset {remote_asset_data.identifier} in the '
+                        f'DB during the v{version} assets update. Skipping entry. '
+                        f'Error: {str(e)}',
+                    )
+                return  # fail or succeed continue to next entry
+
+            if local_asset is not None:
+                AssetResolver().clean_memory_cache(local_asset.identifier.lower())
+
+            # otherwise asset is known, so it's a conflict. Check if we can resolve
+            resolution = conflicts.get(local_asset) if conflicts else None
+            if resolution == 'local':
+                # do nothing, keep local
+                return
+            if resolution == 'remote':
+                try:
+                    with connection.savepoint_ctx() as cursor:
+                        _force_remote(cursor, local_asset, full_insert)
+                except sqlite3.Error as e:
+                    self.msg_aggregator.add_warning(
+                        f'Failed to resolve conflict for {remote_asset_data.identifier} in '
+                        f'the DB during the v{version} assets update. Skipping entry. '
+                        f'Error: {str(e)}',
+                    )
+                return  # fail or succeed continue to next entry
+
+            # else can't resolve. Mark it for the user to resolve.
+            # TODO: When assets refactor is finished, remove the usage of AssetData here
+            local_data = GlobalDBHandler().get_all_asset_data(  # pylint: disable=unsubscriptable-object  # noqa: E501
+                mapping=False,
+                serialized=False,
+                specific_ids=[local_asset.identifier],
+            )[0]
+            self.conflicts.append((local_data, remote_asset_data))
+
     def _apply_single_version_update(
             self,
-            connection: DBConnection,
+            connection: 'DBConnection',
             version: int,
             text: str,
             conflicts: Optional[dict[Asset, Literal['remote', 'local']]],
```

Comment:
I would rename `conflicts` (also `self.conflicts`) if it is only for assets.

Upd: also, for example, `force_remote`, and generally all other only-assets methods